microsetta_public_api/resources_alt.py

```diff
@@ -0,0 +1,99 @@
+from typing import Dict
+
+from microsetta_public_api.backend import constructors
+from microsetta_public_api.config import (
+    SERVER_CONFIG,
+)
+
+
+class Component:
+
+    def __init__(self, name):
+        self.name = name
+        self.data = None
+        self.children: Dict[str, 'Component'] = dict()
+        self.parent = None
+
+    def add_child(self, child: 'Component'):
+        self.children.update({child.name: child})
+        return child
+
+    def set(self, other: 'Component'):
+        self.__dict__ = other.__dict__
+
+    def set_data(self, data: object):
+        self.data = data
+
+    def get_data(self):
+        return self.data
+
+    def get_child(self, name: str) -> 'Component':
+        try:
+            return self.children[name]
+        except KeyError:
+            raise KeyError(name)
+
+    def gets(self, *args):
+        if len(args) == 0:
+            return self
+        first = args[0]
+        rest = args[1:]
+        child = self.get_child(first)
+        if rest:
+            return child.gets(*rest)
+        return child
+
+    def has(self, *args):
+        if len(args) == 0:
+            return True
+        first = args[0]
+        rest = args[1:]
+        if not rest:
+            return first in self.children
+        elif first in self.children:
+            return self.get_child(first).has(*rest)
+        else:
+            return False
+
+    def remove(self, name: str):
```

Comment:
Method name is inconsistent with add_child.  Should be called remove_child maybe?

Reply:
Fixed in https://github.com/biocore/microsetta-public-api/pull/57/commits/bf278b80b5b898af365f5590f1bf2d8ca7254b32